### PR TITLE
merge master into feature

### PR DESCRIPTION
**Pull request by tockards**
This PR merges master into the katstore feature branch

### CODE REVIEW

File: app/sensor-graph/sensor-graph.js [hunk 1]
```diff
@@ -82,6 +85,21 @@
           }
         ];
 
+        vm.getNewKatstoreLink = function () {
```
Comment (ajoubertza) [inline app/sensor-graph/sensor-graph.js]: This came in on previous PR, but looking at it now, "New" doesn't seem like a good name.

File: app/sensor-graph/sensor-graph.js [hunk 2]
```diff
@@ -16,15 +16,18 @@
...
+        vm.showTips = false;
         vm.showContextZoom = true;
         vm.showRelativeTime = false;
         vm.liveData = false;
         vm.useFixedYAxis = false;
         vm.useFixedXAxis = false;
         vm.yAxisMinValue = 0;
         vm.yAxisMaxValue = 100;
-
+        vm.clientSubject = 'katgui.sensor_graph.' + KatGuiUtil.generateUUID();
```
Comment (ajoubertza) [inline app/sensor-graph/sensor-graph.js]: Usage of this was removed in the feature branch previously.  Looks like it was used to get (old) katstore query results published over websocket.

Comment (tockards) [reply app/sensor-graph/sensor-graph.js]: Thanks for the review. I think I should ping you after I got all these branches up on a dev box. As I should run into a lot of these merge conflicts that weren't resolved correctly.

**Comment by tockards** [comment]
I am happy with the changes that I made  based on the comments and requests and happy with the resolved merged conflicts there might be issues in the resolved merged conflicts and those who know katgui with better understanding than me or done work on katgui prior to this PR please spare your time as I would appreciate it if you could review this. 

CC @ajoubertza @bxaia @xinyuwu @bngcebetsha and anyone else who done changes to katgui prior to this PR .

All can be tested on the devq node for verification. 

I have also attached images of the 'link to flap' changes I have made. (bare in mind that I tested the flap changes with a committed katcamconfig branch that added a link to the conf and will be removed from the katcamconfig to remove the link from dev nodes.) 
So you might not see it when you test it on the devq system.

Screenshot 2019-07-24 at 13 31 32

Links to process control without katstore processes(this is more a katcamconfig change but for gui purposes it reflects on this branch)
Screenshot 2019-07-24 at 13 30 44
Screenshot 2019-07-24 at 13 30 57

Testing is also required on sensorgraph and that can be done on devq.

**Pull request merged**